src/grpc/headers-interceptor.ts

```diff
@@ -36,10 +37,12 @@ export class HeaderInterceptor {
     return (options, nextCall) => {
       return new InterceptingCall(nextCall(options), {
         start: (metadata, listener, next) => {
+          metadata.add('request-id', randomUUID());
           this.headersToAddEveryTime.forEach(h =>
             metadata.add(h.name, h.value)
           );
           if (!HeaderInterceptor.areOnlyOnceHeadersSent) {
+            HeaderInterceptor.areOnlyOnceHeadersSent = true;
```

Comment:
this wasn't being set before.  That meant that the agent header was being sent on every request.  I "fixed" it but have mixed feelings about it so would be open to discussing whether this is actually the behavior we want here.

Reply:
It was performance optimization to cut down on bytes sent. TBH feel its bit of pre-mature optimization see comment on request-id header. That being said I dont care about this one as much vs request-ids so would not be end of world if we keep on only first request like it is now.

Reply:
need others to weigh in here.  honestly i'd prefer to send this on every request as well.  I'm pretty sure I know what Kenny's opinion is.  Would like to know where @danielamiao , @mjindalmotiv8 , anyone else who is interested in weighing in on this stand.

Reply:
I think there is value in this during debugging and having the feature to enable it on per customer selectively during debugging is most likely what we should be doing. Doing on each request is going to add more processing on mr2 and given that we care about microseconds worth of operations, I am worried this will add latency.

Reply:
configurability is not in scope for this PR, so I will leave this as-is, meaning that it will only be sent once per client.